lib/ansible/modules/network/iosxr/iosxr_config.py

```diff
@@ -214,46 +246,93 @@ def check_args(module, warnings):
                         'removed in the future')
 
 
+# A list of commands like {end-set, end-policy, ...} are part of configuration
+# block like { prefix-set, as-path-set , ... } but they are not indented properly
+# to be included with their parent. sanitize_config will add indentation to
+# end-* commands so they are included with their parents
+def sanitize_config(config, force_diff_prefix=None):
+    conf_lines = to_native(config, errors='surrogate_or_strict').split('\n')
```

Comment:
Is this conversion required? The input is in text format and the response received from the remote host is converted to text and passed to the module code